app/middleware/renderer.js

```diff
@@ -15,7 +18,36 @@ function age(req, res) {
 }
 
 function choose(req, res) {
-  res.render('choose');
+  return res.render('choose');
+}
+
+function location(req, res) {
+  if (!res.locals.correctLocationParams) {
+    return res.render('error');
+  }
+  return res.render('location');
+}
+
+function results(req, res) {
+  res.render('results');
+}
+
+function emptyPostcode(req, res) {
+  log.debug('Empty search');
+  res.locals.errorMessage = messages.emptyPostcodeMessage();
+  location(req, res);
+}
+
+function invalidPostcode(req, res, loc) {
+  log.debug({ loc }, 'Location failed validation');
+  res.locals.errorMessage = messages.invalidPostcodeMessage(loc);
+  location(req, res);
+}
+
+function outsideOfEngland(req, res, loc) {
+  log.debug({ loc }, 'Outside of England');
```

Comment:
`loc` is the response object from Postcodes.io. Can this be made clear or if that isn't the intention (as it has already been logged in the [postcodeLookup](https://github.com/nhsuk/sexual-health-service-finder/pull/11/files#diff-ea9d7600f9dd35965091987f6fa94e04R38) can jus the location of the request be logged (that isn't included in Postcodes.io response).